api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -686,6 +686,85 @@ private String getQualifiedColumnName(Domain currentDomain, String columnName) {
         .collect(Collectors.toList());
   }
 
+  @Override
+  public List<String> generateMicroarrayCohortExtractCodeCells(
+      DbWorkspace dbWorkspace,
+      String qualifier,
+      Map<String, QueryJobConfiguration> queriesByDomain) {
+    String joinedDatasetVariableNames =
+        queriesByDomain.entrySet().stream()
+            .map(
+                e ->
+                    "dataset_"
+                        + qualifier
+                        + "_"
+                        + Domain.fromValue(e.getKey()).toString().toLowerCase()
+                        + "_df")
+            .collect(Collectors.joining(", "));
+
+    return ImmutableList.of(
+        "person_ids = set()\n"
+            + "datasets = ["
+            + joinedDatasetVariableNames
+            + "]\n"
+            + "\n"
+            + "for dataset in datasets:\n"
+            + "    if 'PERSON_ID' in dataset:\n"
+            + "        person_ids = person_ids.union(dataset['PERSON_ID'])\n"
+            + "    elif 'person_id' in dataset:\n"
+            + "        person_ids = person_ids.union(dataset['person_id']) \n"
+            + "\n\n"
+            + "with open('cohort_sample_names.txt', 'w') as cohort_file:\n"
+            + "    for person_id in person_ids:\n"
+            + "        cohort_file.write(str(person_id) + '\\n')\n"
+            + "    cohort_file.close()\n",
+        "%%bash\n\n"
+            + "uuid=$(cat /proc/sys/kernel/random/uuid | sed s/-/_/g)\n"
+            + "EXPORT_TABLE=\"fc-aou-cdr-synth-test.tmp_shared_cohort_extract.${uuid}\"\n"
+            + "\n"
+            + "python3 /genomics/microarray/raw_array_cohort_extract.py \\\n"
+            + "          --dataset fc-aou-cdr-synth-test.microarray_data \\\n"
+            + "          --fq_destination_table ${EXPORT_TABLE} \\\n"
+            + "          --query_project "
+            + dbWorkspace.getWorkspaceNamespace()
+            + " \\\n"
+            + "          --sample_mapping_table fc-aou-cdr-synth-test.microarray_data.sample_list \\\n"
+            + "          --cohort_sample_names_file cohort_sample_names.txt \\\n"
+            + "          --sample_map_outfile cohort_sample_map.csv\n"
+            + "\n"
+            + "gatk ArrayExtractCohort \\\n"
+            + "        -R /genomics/Homo_sapiens_assembly19.fasta \\\n"
```

Comment:
Assuming it is mounted at /genomics sort of forces us into putting it on the image, or localizing the reference fasta in the startup script.

Have you tried just supplying a gs:// path to this flag?

Reply:
Still curious about this - same for probe_info